main/ZmqttDiscovery.ino

```diff
@@ -125,7 +127,32 @@ void createDiscovery(char* sensor_type,
     device.set("sw_version", OMG_VERSION);
     JsonArray& identifiers = device.createNestedArray("identifiers");
     identifiers.add(getMacAddress());
-    sensor.set("device", device); //device sensor is connected to
+    sensor.set("device", device); //device representing the board
+  } else {
+    char deviceid[13];
+    memcpy(deviceid, &unique_id[0], 12);
+    deviceid[12] = '\0';
+    StaticJsonBuffer<JSON_MSG_BUFFER> jsonDeviceBuffer;
+    JsonObject& device = jsonDeviceBuffer.createObject();
+    if (device_mac != "") {
+      JsonArray& connections = device.createNestedArray("connections");
+      JsonArray& connection_mac = connections.createNestedArray();
+      connection_mac.add("mac");
+      connection_mac.add(device_mac);
+    }
+    JsonArray& identifiers = device.createNestedArray("identifiers");
+    identifiers.add(deviceid);
+    if (device_manufacturer != "") {
+      device.set("manufacturer", device_manufacturer);
+    }
+    if (device_model != "") {
+      device.set("model", device_model);
+    }
+    if (device_name != "") {
```

Comment:
This check against a char array will not work, you should use this kind of expression:
` if (cmd_topic[0]) {`

Reply:
I fixed it but I'm not sure about this one.